src/use-ref-value.js

```diff
@@ -2,6 +2,14 @@
  * @typedef {import('./style-dictionary-to-figma.js').Obj} Obj
  */
 
+/**
+ * @param {string} key
+ * @returns {boolean}
+ */
+function isRefValue(key) {
+  return key.startsWith('{') && key.endsWith('}');
```

Comment:
value can be of different types - number, array, object - these need special handling, right?

Reply:
Object example:
```
{
  "font": {
    "body": {
      "regular": {
        "value": {
          "fontFamily": "{fontFamily.body}",
          "fontWeight": "{fontWeight.regular}",
          "lineHeight": "{size.lineHeight.xsmall}",
          "fontSize": "{size.font.xsmall}"
        }
      }
    }
  }
}
```

Array example:

```
{
  "shadow": {
    "elevation": {
      "small": {
        "value": ["{shadow.core.4}", "{shadow.core.2}"]
      }
    }
  }
}
```

Probably needs to be recursive as well to be safe - for example if the shadow token did not use references, it would be an array of objects. Not sure how common deeper arrays/objects are.

Reply:
True I guess, I'll add another case for this and expand this function to check for nested array/obj

Reply:
So, I think I covered these cases now recursively, the only issue is that if a value is an object or array consisting of multiple nested values, I can't really handle easily cases where some of the props/items are references and some aren't. If even 1 prop/item is not a reference, I'm handling it as if the entire parent value is "not a reference" and it won't restore the original value. 

Perhaps in the future it could be smarter about going into the original value and restoring pieces of it that are references, but this would definitely complicate the transformer a fair bit I think, not sure how important it is for users (e.g. you) right now?

Reply:
Hmm, yeah this is definitely a more advanced usecase and not too important. My specific issue was also on tokens that are strings, most of the complicated tokens with arrays/objects use references anyway.

I am thinking though that maybe it should default to the opposite - that even if one prop/item is a reference, restore the original value with references? That way, if there are still issues with some original value being wrong, it can still be disabled manually with `ignoreUseRefValue`, but in most cases I think it makes sense to restore the references.

Reply:
That does make sense indeed. Adjusted.